### PR TITLE
drivers: gnss: Place API into iterable section

### PR DESCRIPTION
**Pull request by pdgendt**
Add wrapper DEVICE_API macro to all gnss_driver_api instances.

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]

**Review by tomi-font — DISMISSED** [review]

File: include/zephyr/drivers/gnss.h [hunk 1]
```diff
@@ -237,13 +237,13 @@ __syscall int gnss_set_fix_rate(const struct device *dev, uint32_t fix_interval_
 
 static inline int z_impl_gnss_set_fix_rate(const struct device *dev, uint32_t fix_interval_ms)
 {
-	const struct gnss_driver_api *api = (const struct gnss_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(gnss, dev));
```
[the remark this answers is not in the document]
Comment (tomi-font) [reply include/zephyr/drivers/gnss.h:240]: Of course you do. But to a reasonable extent such that you can still enable debug features. :wink: I don't know, I'm just saying this without knowing numbers. `ASSERT` is already better than `CHECKIF`, but I'm wondering how much value it still really has.

[the remark this answers is not in the document]
Comment (pdgendt) [reply include/zephyr/drivers/gnss.h:240]: The assertion message (which stringifies the assertion test) should result in the same string each time, so I'd assume the compiler is smart enough to only have a single occurrence in the resulting binary.

[the remark this answers is not in the document]
Comment (pdgendt) [reply include/zephyr/drivers/gnss.h:240]: For `gcc` we need `-fmerge-constants`, I think, which is enabled in `-O1`.

[the remark this answers is not in the document]
Comment (bjarki-andreasen) [reply include/zephyr/drivers/gnss.h:240]: Also, note that the wrappers are static inline, so if they are not used by the application, the assert is not compiled in :) most applications will likely only use a subset of all of the available APIs

[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply include/zephyr/drivers/gnss.h:240]: > so I'd assume the compiler is smart enough to only have a single occurrence in the resulting binary.

It does, one thing I always had on the back of my mind is coming up with some standard error strings for common errors so we don't have a million variation of "register read fail" or "device X init fail" in the binary, broader issue though

[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply include/zephyr/drivers/gnss.h:240]: wow I just did that for something I have kicking around and realized `BT_ASSERT` is on by default and disabling it saved a whole 15Kb of flash. Made my day thanks for the comment @tomi-font :-)

**Review by fabiobaltieri — APPROVED** [review]

**Review by tomi-font — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**